lib/media_types/scheme.rb

```diff
@@ -385,24 +396,57 @@ def inspect(indentation = 0)
     end
 
     def assert_pass(fixture)
-      json = JSON.parse(fixture, { symbolize_names: true })
-
-      validate(json)
+      caller = caller_locations[1]
+      object_to_store = {
+        fixture: fixture,
+        expect_to_pass: true,
+        caller: caller
+        }
+      @fixtures << object_to_store
     end
-    
+
     def assert_fail(fixture)
-      json = JSON.parse(fixture, { symbolize_names: true })
+      caller = caller_locations[1]
+      object_to_store = {
+        fixture: fixture,
+        expect_to_pass: false,
+        caller: caller
+        }
+      @fixtures << object_to_store
+    end
+
+    def execute_assertions(media_type_class)
```

Comment:
```suggestion
    def execute_assertions
```

Argument is not used